glommio/src/executor/mod.rs

```diff
@@ -2338,6 +2340,69 @@ impl ExecutorProxy {
                 .map(|x| ScopedTask::<'a, T>(x, PhantomData))
         })
     }
+
+    /// Spawns a blocking task into a background thread where blocking is
+    /// acceptable.
+    ///
+    /// Glommio depends on cooperation from tasks in order to drive IO and meet
+    /// latency requirements. Unyielding tasks are detrimental to the
+    /// performance of the overall system, not just to the performance of
+    /// the one stalling task.
+    ///
+    /// `spawn_blocking` is there as a last resort when a blocking task needs to
+    /// be executed and cannot be made cooperative. Examples are:
+    /// * Expensive syscalls that cannot use `io_uring`, such as `mmap`
+    ///   (especially with `MAP_POPULATE`)
+    /// * Calls to synchronous third-party code (compression, encoding, etc.)
+    ///
+    /// It is recommended to guard against calling this method if the blocking
+    /// operation is expected to be fast. For instance, calling `mmap` over a
+    /// small size can be safely called directly in asynchronous code.
+    ///
+    /// # Note
+    ///
+    /// *This method is not meant to be a way to achieve compute parallelism.*
+    /// Distributing work across executors is the better way to achieve that.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use glommio::{LocalExecutor, Task};
+    /// use std::time::Duration;
+    ///
+    /// let local_ex = LocalExecutor::default();
+    ///
+    /// local_ex.run(async {
+    ///     let task = glommio::executor()
+    ///         .spawn_blocking(|| {
+    ///             std::thread::sleep(Duration::from_millis(100));
+    ///         })
+    ///         .await;
+    /// });
+    /// ```
+    pub fn spawn_blocking<F, R>(&self, func: F) -> impl Future<Output = R>
+    where
+        F: FnOnce() -> R + Send + 'static,
+        R: Send + 'static,
+    {
+        let result = Arc::new(Mutex::new(MaybeUninit::<R>::uninit()));
```

Comment:
this overall looks pretty good but I am not sure I like this mutex.

I suspect we could use `from_raw()` and `into_raw()` and pass the pointer. If not, ultimately it is fine.